cpp/src/slice2cs/Gen.cpp

```diff
@@ -1023,13 +1014,13 @@ Slice::CsVisitor::writeValue(const TypePtr& type, const string& ns)
     StructPtr st = dynamic_pointer_cast<Struct>(type);
     if (st)
     {
-        if (st->hasMetaData("cs:class"))
+        if (isMappedToClass(st))
         {
             return "null";
         }
         else
         {
-            return "new " + typeToString(type, ns) + "()";
+            return "default";
         }
     }
```

Comment:
Can we use `default` for both branches here too?

Reply:
I simplified the code. 

If you look at the whole function, I think the idea is for the generated value to be readable, with 0 / default / null as appropriate.